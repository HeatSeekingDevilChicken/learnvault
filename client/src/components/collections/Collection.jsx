import React from 'react';
import { Link } from 'react-router-dom';

const Collection = ({
  id, title, description, author,
}) => {
  function likeButtonClick(eventId) {
    console.log('ID of collection that was liked == ', eventId);
  }

  return (

    <div key={id} className="collection-div">
      <h1>
        Title:
        {' '}
        {' '}
        {title}
      </h1>
      <h3>
        Description:
        {' '}
        {' '}
        {description}
      </h3>

      <p>
        Creator:
        {' '}
        {' '}
        {author}
        {' '}
        {' '}
      </p>
      <div>
        <Link to={`/collections/${id}`}>View Collection</Link>
      </div>
      <div>
        <button type="button" onClick={() => likeButtonClick(id)}>Like</button>
      </div>
    </div>

  );
};

export default Collection;
