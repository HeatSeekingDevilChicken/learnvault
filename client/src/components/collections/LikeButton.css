.button-like {
  margin-right: 10px;
}