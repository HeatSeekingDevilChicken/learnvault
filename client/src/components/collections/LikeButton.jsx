import React from 'react';

const LikeButton = ({ id, loggedInUser }) => {
  function likeButtonClick(eventId, userId) {
    console.log('ID of collection that was liked == ', eventId);
    console.log('ID of user that clicked like == ', userId);

    const payload = { id: userId, collectionId: id };

    fetch(`/api/collections/like/${id}`, {
      method: 'PUT', // or 'PUT'
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify(payload),
    })
      .then((response) => response.json())
      .then((data) => {
        console.log('Success:', data);
      })
      .catch((error) => {
        console.error('Error:', error);
      });
  }

  return (
    <button label="likebutton" onClick={() => likeButtonClick(id, loggedInUser)} type="button" className="like-button">
      <i className="far fa-thumbs-up" />
    &nbsp; Like Collection
    </button>
  );
};

export default LikeButton;
