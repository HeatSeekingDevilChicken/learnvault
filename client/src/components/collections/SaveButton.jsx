import React from 'react';

const SaveButton = ({ id, loggedInUser }) => {
  function likeButtonClick(eventId, userId) {
    const payload = { id: userId, collectionId: id };

    fetch(`/api/collections/save/${id}`, {
      method: 'PUT',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify(payload),
    })
      .then((response) => response.json())
      .then((data) => {
        if (process.env.NODE_ENV === 'development'
          || process.env.NODE_ENV === 'test') {
          console.log('Success:', data);
        }
        return data;
      })
      .catch((error) => {
        if (process.env.NODE_ENV === 'development'
          || process.env.NODE_ENV === 'test') {
          console.error('Error:', error);
        }
      });
  }

  return (
    <span
      onClick={() => likeButtonClick(id, loggedInUser)}
      onKeyPress={() => likeButtonClick(id, loggedInUser)}
      role="button"
      tabIndex={0}
      type="button"
      className="button-like"
    >
      <i className="far fa-star" />
    &nbsp; Save Collection
    </span>
  );
};

export default SaveButton;
