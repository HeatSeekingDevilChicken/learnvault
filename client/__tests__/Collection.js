/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import LikeButton from '../src/components/collections/LikeButton';
import SaveButton from '../src/components/collections/SaveButton';
import Collection from '../src/components/collections/Collection';

configure({ adapter: new Adapter() });

describe('Collection tests', () => {
  describe('Collection', () => {
    let wrapper;
    const props = {
      key: 544,
      id: 4354,
      title: 'Collection',
      description: 'buncha stuff',
      author: 'Veronica',
      loggedInUser: 'Sarge',
    };

    beforeAll(() => {
      wrapper = shallow(
        <Collection
          key={props.key}
          id={props.id}
          title={props.title}
          description={props.description}
          author={props.author}
          loggedInUser={props.loggedInUser}
        />,
      );
    });

    it('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('renders an h1 element that displays collection title', () => {
      expect(wrapper.find('h1').text()).toMatch(props.title);
    });

    it('renders an h3 element that displays collection description', () => {
      expect(wrapper.find('h3').text()).toMatch(props.description);
    });

    it('displays div with name of author', () => {
      expect(wrapper.find('div').first().text()).toMatch(props.author);
    });

    it('displays a Link with view the collection', () => {
      expect(wrapper.find('Link').text()).toMatch('View Collection');
    });

    it('displays Like and Save buttons if logged in', () => {
      expect(wrapper.find('LikeButton')).toHaveLength(1);
      expect(wrapper.find('SaveButton')).toHaveLength(1);
    });

    // alternate way to actually verify the components are being rendered
    // uses package: jest-enzyme
    it('displays Like and Save buttons if logged in', () => {
      expect(wrapper).toContainReact(<LikeButton loggedInUser={props.loggedInUser} id={props.id} />);
      expect(wrapper).toContainReact(<SaveButton loggedInUser={props.loggedInUser} id={props.id} />);
    });

    it('displays message to register or login if not logged in', () => {
      wrapper = shallow(
        <Collection
          key={props.key}
          id={props.id}
          title={props.title}
          description={props.description}
          author={props.author}
          loggedInUser={null}
        />,
      );
      expect(wrapper.text()).toMatch('Register');
      expect(wrapper.text()).toMatch('Login');
    });
  });
});
