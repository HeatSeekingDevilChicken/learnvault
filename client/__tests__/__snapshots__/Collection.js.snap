// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collection tests Collection renders correctly 1`] = `
<RenderWithRouter>
  <MemoryRouter
    initialEntries={
      Array [
        Object {
          "key": "testKey",
          "pathname": "/",
        },
      ]
    }
  >
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "key": "testKey",
              "pathname": "/",
              "search": "",
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "testKey",
            "pathname": "/",
            "search": "",
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Route
        path="/"
      >
        <Collection
          author="Veronica"
          description="buncha stuff"
          id="5eee5bac1e986de551d57488"
          key="testKey"
          likes={
            Array [
              "5ef2b8c3d5973033a191aea2",
              "5ef3f1798a8800471b987bbe",
              "5ef7840ba4aca16282b26299",
            ]
          }
          loggedInUser="Sarge"
          title="Collection"
        >
          <div
            className="collection"
            key="5eee5bac1e986de551d57488"
          >
            <h1
              className="collection__title"
            >
              Collection
            </h1>
            <h3>
              buncha stuff
            </h3>
            <div
              className="creator"
            >
              <div
                className="creator__label"
              >
                Creator:
              </div>
              <div
                className="creator__author"
              >
                Veronica
              </div>
            </div>
            <div>
              <WithStyles(ForwardRef(Button))
                color="primary"
                component={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "displayName": "Link",
                    "propTypes": Object {
                      "innerRef": [Function],
                      "onClick": [Function],
                      "replace": [Function],
                      "target": [Function],
                      "to": [Function],
                    },
                    "render": [Function],
                  }
                }
                to="/collections/5eee5bac1e986de551d57488"
                variant="contained"
              >
                <ForwardRef(Button)
                  classes={
                    Object {
                      "colorInherit": "MuiButton-colorInherit",
                      "contained": "MuiButton-contained",
                      "containedPrimary": "MuiButton-containedPrimary",
                      "containedSecondary": "MuiButton-containedSecondary",
                      "containedSizeLarge": "MuiButton-containedSizeLarge",
                      "containedSizeSmall": "MuiButton-containedSizeSmall",
                      "disableElevation": "MuiButton-disableElevation",
                      "disabled": "Mui-disabled",
                      "endIcon": "MuiButton-endIcon",
                      "focusVisible": "Mui-focusVisible",
                      "fullWidth": "MuiButton-fullWidth",
                      "iconSizeLarge": "MuiButton-iconSizeLarge",
                      "iconSizeMedium": "MuiButton-iconSizeMedium",
                      "iconSizeSmall": "MuiButton-iconSizeSmall",
                      "label": "MuiButton-label",
                      "outlined": "MuiButton-outlined",
                      "outlinedPrimary": "MuiButton-outlinedPrimary",
                      "outlinedSecondary": "MuiButton-outlinedSecondary",
                      "outlinedSizeLarge": "MuiButton-outlinedSizeLarge",
                      "outlinedSizeSmall": "MuiButton-outlinedSizeSmall",
                      "root": "MuiButton-root",
                      "sizeLarge": "MuiButton-sizeLarge",
                      "sizeSmall": "MuiButton-sizeSmall",
                      "startIcon": "MuiButton-startIcon",
                      "text": "MuiButton-text",
                      "textPrimary": "MuiButton-textPrimary",
                      "textSecondary": "MuiButton-textSecondary",
                      "textSizeLarge": "MuiButton-textSizeLarge",
                      "textSizeSmall": "MuiButton-textSizeSmall",
                    }
                  }
                  color="primary"
                  component={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "displayName": "Link",
                      "propTypes": Object {
                        "innerRef": [Function],
                        "onClick": [Function],
                        "replace": [Function],
                        "target": [Function],
                        "to": [Function],
                      },
                      "render": [Function],
                    }
                  }
                  to="/collections/5eee5bac1e986de551d57488"
                  variant="contained"
                >
                  <WithStyles(ForwardRef(ButtonBase))
                    className="MuiButton-root MuiButton-contained MuiButton-containedPrimary"
                    component={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "displayName": "Link",
                        "propTypes": Object {
                          "innerRef": [Function],
                          "onClick": [Function],
                          "replace": [Function],
                          "target": [Function],
                          "to": [Function],
                        },
                        "render": [Function],
                      }
                    }
                    disabled={false}
                    focusRipple={true}
                    focusVisibleClassName="Mui-focusVisible"
                    to="/collections/5eee5bac1e986de551d57488"
                    type="button"
                  >
                    <ForwardRef(ButtonBase)
                      className="MuiButton-root MuiButton-contained MuiButton-containedPrimary"
                      classes={
                        Object {
                          "disabled": "Mui-disabled",
                          "focusVisible": "Mui-focusVisible",
                          "root": "MuiButtonBase-root",
                        }
                      }
                      component={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "displayName": "Link",
                          "propTypes": Object {
                            "innerRef": [Function],
                            "onClick": [Function],
                            "replace": [Function],
                            "target": [Function],
                            "to": [Function],
                          },
                          "render": [Function],
                        }
                      }
                      disabled={false}
                      focusRipple={true}
                      focusVisibleClassName="Mui-focusVisible"
                      to="/collections/5eee5bac1e986de551d57488"
                      type="button"
                    >
                      <Link
                        aria-disabled={false}
                        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
                        onBlur={[Function]}
                        onDragLeave={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        onKeyUp={[Function]}
                        onMouseDown={[Function]}
                        onMouseLeave={[Function]}
                        onMouseUp={[Function]}
                        onTouchEnd={[Function]}
                        onTouchMove={[Function]}
                        onTouchStart={[Function]}
                        role="button"
                        tabIndex={0}
                        to="/collections/5eee5bac1e986de551d57488"
                      >
                        <LinkAnchor
                          aria-disabled={false}
                          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
                          href="/collections/5eee5bac1e986de551d57488"
                          navigate={[Function]}
                          onBlur={[Function]}
                          onDragLeave={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          onKeyUp={[Function]}
                          onMouseDown={[Function]}
                          onMouseLeave={[Function]}
                          onMouseUp={[Function]}
                          onTouchEnd={[Function]}
                          onTouchMove={[Function]}
                          onTouchStart={[Function]}
                          role="button"
                          tabIndex={0}
                        >
                          <a
                            aria-disabled={false}
                            className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
                            href="/collections/5eee5bac1e986de551d57488"
                            onBlur={[Function]}
                            onClick={[Function]}
                            onDragLeave={[Function]}
                            onFocus={[Function]}
                            onKeyDown={[Function]}
                            onKeyUp={[Function]}
                            onMouseDown={[Function]}
                            onMouseLeave={[Function]}
                            onMouseUp={[Function]}
                            onTouchEnd={[Function]}
                            onTouchMove={[Function]}
                            onTouchStart={[Function]}
                            role="button"
                            tabIndex={0}
                          >
                            <span
                              className="MuiButton-label"
                            >
                              View Collection
                            </span>
                          </a>
                        </LinkAnchor>
                      </Link>
                    </ForwardRef(ButtonBase)>
                  </WithStyles(ForwardRef(ButtonBase))>
                </ForwardRef(Button)>
              </WithStyles(ForwardRef(Button))>
            </div>
            <div>
              <br />
              <LikeButton
                id="5eee5bac1e986de551d57488"
                likes={
                  Array [
                    "5ef2b8c3d5973033a191aea2",
                    "5ef3f1798a8800471b987bbe",
                    "5ef7840ba4aca16282b26299",
                  ]
                }
                loggedInUser="Sarge"
              >
                <span
                  className="button-like"
                  onClick={[Function]}
                  onKeyPress={[Function]}
                  role="button"
                  tabIndex={0}
                  type="button"
                >
                  <i
                    className="far fa-thumbs-up"
                  />
                  <small>
                     
                    3
                  </small>
                </span>
              </LikeButton>
              <SaveButton
                id="5eee5bac1e986de551d57488"
                loggedInUser="Sarge"
              >
                <span
                  className="button-like"
                  onClick={[Function]}
                  onKeyPress={[Function]}
                  role="button"
                  tabIndex={0}
                  type="button"
                >
                  <i
                    className="far fa-star"
                  />
                    Save Collection
                </span>
              </SaveButton>
            </div>
          </div>
        </Collection>
      </Route>
    </Router>
  </MemoryRouter>
</RenderWithRouter>
`;
